### PR TITLE
[monitors] LoopMonitor tracks end pc of loop and loop depth

### LINKED ISSUE #10: Loop monitor should show loop nesting
The `--monitors=loops` monitor should track the nesting depth of each loop and reflect that in its output report by indenting each loop according to its nesting level.

### PR DESCRIPTION
**Pull request by btwj**
LoopMonitor keeps track of loop depth and end pc's for loops.

Closes #10

### CODE REVIEW

File: src/monitors/LoopMonitor.v3 [hunk 1]
```diff
@@ -2,22 +2,52 @@
...
+
 			for (bi = it.reset(f); bi.more(); bi.next()) {
-				if (bi.current() == Opcode.LOOP) {
-					var entry = entries.length;
-					m.insertProbeAt(f.func_index, bi.pc, LoopMonitorCounter.new(this, entry));
-					entries.put(m, f, bi.pc);
-					counts.put(0);
+				var op = bi.current();
+				match (op) {
+					BLOCK, IF => controlStack.push(op);
```
Comment (titzer) [inline src/monitors/LoopMonitor.v3:27]: Can you put in a TODO for `try` and exception-handling related bytecodes? Might be worthwhile to add a utility that automatically tracks the control stack in the future too.

File: src/monitors/LoopMonitor.v3 [hunk 2]
```diff
@@ -2,22 +2,52 @@
...
 
 	def onParse(m: Module) {
 		var it = BytecodeIterator.new();
+
 		for (i < m.functions.length) {
 			var f = m.functions[i];
 			if (f.imp != null) continue; // skip imported functions
+
+			var loopDepth = 0;
+			var controlStack = ArrayStack<Opcode>.new();
```
Comment (titzer) [inline src/monitors/LoopMonitor.v3:21]: Can this be one stack of (Opcode, LmEntry)?

**Review by titzer — COMMENTED** [review]
lgtm with minor simplification.

File: src/monitors/LoopMonitor.v3 (hunk 1 above)
Comment (btwj) [reply src/monitors/LoopMonitor.v3]: Added a TODO, will look at making the utility in the future (another PR)?

**Review by titzer — APPROVED** [review]

**Pull request merged**